proposals/1681/naming_schemas.md

```diff
@@ -0,0 +1,90 @@
+---
+Author: Benji Lampel
+Created: 03/07/2023
+Issue: https://github.com/OpenLineage/OpenLineage/issues/1681
+---
+
+**Purpose**
+The Naming.md file should be reworked as a more programmatic solution with clear, specific definitions.
+
+Names and Namespaces are currently slightly nebulous concepts. They can best be described, perhaps, by two components that, together, form a unique URI to a specific dataset for datasets, and a unique name for a Job. This is a serviceable definition for databases, datalakes, and distributed file stores, where a well-defined path exists by nature of the structure. But this may not be the case with database-like systems, or non-database systems, like Salesforce or Google Sheets, respectively. These instances need another way to specify a unique name, one that may not resolve to a URI.
```

Comment:
Is there any example of potential APIs we'd have for this proposals for Python/Java and how we'd use this in integrations?

Reply:
+1 to @mobuchowski

Proposal itself does not modify existing schema and datasets are still identified by namespace and name which are strings. 

It's worth then describing what will be done with naming schemas. These can be: 
 * client builder methods that build namespace out of its components
 * client validation methods
 * Marquez returning additional structured name/namespace